pkg/agent/controller/egress/egress_controller_test.go

```diff
@@ -1118,7 +1118,7 @@ func TestSyncEgress(t *testing.T) {
 			if tt.newLocalIPs != nil {
 				c.localIPDetector = &fakeLocalIPDetector{localIPs: tt.newLocalIPs}
 			}
-			assert.NoError(t, wait.Poll(time.Millisecond*100, time.Second, func() (done bool, err error) {
+			assert.NoError(t, wait.PollUntilContextTimeout(context.Background(), time.Millisecond*100, time.Second, false, func(ctx context.Context) (done bool, err error) {
```

Comment:
this may be a good opportunity to update these to `assert.Eventually` / `require.Eventually`

what do you think @tnqn ?

Taking into account of course this shortcoming: https://github.com/stretchr/testify/issues/1396